src/codeflare_sdk/utils/generate_cert.py

```diff
@@ -74,6 +75,24 @@ def generate_ca_cert(days: int = 30):
     return key, certificate
 
 
+def get_secret_name(cluster_name, namespace, api_instance):
+    label_selector = f"ray.openshift.ai/cluster-name={cluster_name}"
+    try:
+        secrets = api_instance.list_namespaced_secret(
+            namespace, label_selector=label_selector
+        )
+        for secret in secrets.items:
+            if (
+                f"{cluster_name}-ca-secret-" in secret.metadata.name
```

Comment:
Would it have sense to rather select secret based on `ray.openshift.ai/cluster-name` label?

Reply:
It seems, but it's true there can still be multiple secrets for the same cluster?

Reply:
Though, there may be more secrets with the label.

Reply:
The oauth config secret also shares this label so we need to be more specific

Reply:
At a later point, we can add an annotation to the client Secret to tell them apart more robustly.

Reply:
The label selector is already there, I overlooked it.